core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -288,37 +288,10 @@ private void notifySubscriberOfCloseEvent0(SubscriptionImpl subscription, CloseE
             event.notifySubscriber(subscription, whenComplete());
         } finally {
             subscription.clearSubscriber();
-            Throwable cause = event.cause;
             if (state == State.CLEANUP) {
-                cleanupCause = cause;
-            }
-            for (;;) {
-                final Object e = queue.poll();
-                if (e == null) {
-                    break;
-                }
-
-                // We already notified to the subscriber so skip.
-                if (e instanceof CloseEvent) {
-                    continue;
-                }
-
-                if (e instanceof CompletableFuture) {
-                    if (cause == null) {
-                        cause = ClosedStreamException.get();
-                    }
-                    ((CompletableFuture<?>) e).completeExceptionally(cause);
-                    continue;
-                }
-
-                try {
-                    @SuppressWarnings("unchecked")
-                    final T obj = (T) e;
-                    onRemoval(obj);
-                } finally {
-                    StreamMessageUtil.closeOrAbort(e, cause);
-                }
+                cleanupCause = event.cause;
             }
+            cleanupObjects(event.cause);
```

Comment:
- nit: Could extract `event.cause` into a local variable like before.
- Thanks a lot for cleaning this up!